### ISSUE TITLE
JTReg Test Failure : java/lang/invoke/condy/CondyNameValidationTest.java

### ISSUE THREAD

**Issue by ben-walsh** [eclipse/openj9#4966]
Consistent failures ...

```
[2019-03-04T10:41:50.928Z] test CondyNameValidationTest.testInvalidNames(""): failure
[2019-03-04T10:41:50.928Z] org.testng.TestException: 
[2019-03-04T10:41:50.928Z] Method CondyNameValidationTest.testInvalidNames(java.lang.String)[pri:0, instance:CondyNameValidationTest@7d95506] should have thrown an exception of type class java.lang.ClassFormatError
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:89)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.handleInvocationResults(Invoker.java:1384)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:677)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:773)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestRunner.run(TestRunner.java:623)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestNG.run(TestNG.java:1018)
[2019-03-04T10:41:50.928Z] 	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
[2019-03-04T10:41:50.928Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-03-04T10:41:50.928Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-03-04T10:41:50.928Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-03-04T10:41:50.928Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-03-04T10:41:50.928Z] 	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:298)
[2019-03-04T10:41:50.928Z] 	at java.base/java.lang.Thread.run(Thread.java:825)
[2019-03-04T10:41:50.928Z] test CondyNameValidationTest.testInvalidNames("."): failure
[2019-03-04T10:41:50.928Z] org.testng.TestException: 
[2019-03-04T10:41:50.928Z] Method CondyNameValidationTest.testInvalidNames(java.lang.String)[pri:0, instance:CondyNameValidationTest@7d95506] should have thrown an exception of type class java.lang.ClassFormatError
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:89)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.handleInvocationResults(Invoker.java:1384)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:677)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2019-03-04T10:41:50.928Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:773)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestRunner.run(TestRunner.java:623)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2019-03-04T10:41:50.928Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
[2019-03-04T10:41:50.928Z] 	at org.testng.TestNG.run(TestNG.java:1018)
[2019-03-04T10:41:50.928Z] 	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
[2019-03-04T10:41:50.928Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-03-04T10:41:50.928Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-03-04T10:41:50.928Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:298)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.Thread.run(Thread.java:825)
[2019-03-04T10:41:50.929Z] test CondyNameValidationTest.testInvalidNames(";"): failure
[2019-03-04T10:41:50.929Z] org.testng.TestException: 
[2019-03-04T10:41:50.929Z] Method CondyNameValidationTest.testInvalidNames(java.lang.String)[pri:0, instance:CondyNameValidationTest@7d95506] should have thrown an exception of type class java.lang.ClassFormatError
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:89)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.handleInvocationResults(Invoker.java:1384)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:677)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:773)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestRunner.run(TestRunner.java:623)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.run(TestNG.java:1018)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:298)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.Thread.run(Thread.java:825)
[2019-03-04T10:41:50.929Z] test CondyNameValidationTest.testInvalidNames("["): failure
[2019-03-04T10:41:50.929Z] org.testng.TestException: 
[2019-03-04T10:41:50.929Z] Method CondyNameValidationTest.testInvalidNames(java.lang.String)[pri:0, instance:CondyNameValidationTest@7d95506] should have thrown an exception of type class java.lang.ClassFormatError
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:89)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.handleInvocationResults(Invoker.java:1384)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:677)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:773)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestRunner.run(TestRunner.java:623)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.run(TestNG.java:1018)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:298)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.Thread.run(Thread.java:825)
[2019-03-04T10:41:50.929Z] test CondyNameValidationTest.testInvalidNames("/"): failure
[2019-03-04T10:41:50.929Z] org.testng.TestException: 
[2019-03-04T10:41:50.929Z] Method CondyNameValidationTest.testInvalidNames(java.lang.String)[pri:0, instance:CondyNameValidationTest@7d95506] should have thrown an exception of type class java.lang.ClassFormatError
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.ExpectedExceptionsHolder.noException(ExpectedExceptionsHolder.java:89)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.handleInvocationResults(Invoker.java:1384)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:677)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:821)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1131)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2019-03-04T10:41:50.929Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:773)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestRunner.run(TestRunner.java:623)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2019-03-04T10:41:50.929Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1185)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1110)
[2019-03-04T10:41:50.929Z] 	at org.testng.TestNG.run(TestNG.java:1018)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.TestNGRunner.main(TestNGRunner.java:94)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-03-04T10:41:50.929Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[2019-03-04T10:41:50.929Z] 	at com.sun.javatest.regtest.agent.MainActionHelper$AgentVMRunnable.run(MainActionHelper.java:298)
[2019-03-04T10:41:50.929Z] 	at java.base/java.lang.Thread.run(Thread.java:825)
```

... observed with test https://github.com/ibmruntimes/openj9-openjdk-jdk12/blob/openj9/test/jdk/java/lang/invoke/condy/CondyNameValidationTest.java against an OpenJDK12+OpenJ9 build (https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk12u/job/jdk12u-linux-x64-openj9/16/artifact/workspace/target/OpenJDK12U-jdk_x64_linux_openj9_2019-02-28-09-45.tar.gz).

See [consoleText.txt](https://github.com/eclipse/openj9/files/2926904/consoleText.txt).

FYI : Test https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/test/jdk/java/lang/invoke/condy/CondyNameValidationTest.java, which appears identical, shows no such issues against OpenJDK11+OpenJ9 builds - e.g. https://ci.adoptopenjdk.net/view/Test_openjdk/job/openjdk11_j9_openjdktest_x86-64_linux/298/testReport/java_lang_invoke_condy_CondyNameValidationTest/java/CondyNameValidationTest

**Comment by JasonFengJ9** [comment]
Jack is on this.

**Comment by fengxue-IS** [comment]
Trace the code to https://github.com/ibmruntimes/openj9-openjdk-jdk12/blob/openj9/test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java
which calls
`line 140: Class<?> gc = l.defineClass(byteArray);`

Given the known issue with https://github.com/eclipse/openj9/issues/3641, I suspect the test failure is due to unsafe lib skipping the verification process.

**Comment by JasonFengJ9** [comment]
Can we check if https://github.com/eclipse/openj9/pull/4919 (fix for https://github.com/eclipse/openj9/issues/3641) helps this test?

**Comment by JasonFengJ9** [comment]
Verified manually that #4919 makes this test pass.

**Comment by fjeremic** [comment]
> Verified manually that #4919 makes this test pass.

Sounds like we can close this issue then? Or leave it for the author to close once they've confirmed?

**Comment by pshipton** [comment]
@fjeremic I don't think we should close this until #4919 is finalized, merged, and this test is re-tested with the final fix.

**Comment by pshipton** [comment]
#4919 has been merged. 
#5067 should be merged shortly for the equivalent fix in the 0.13 release branch
